### ISSUE TITLE
No valid solidity source files found.

### ISSUE THREAD

**Issue by aliveli186** [ConsenSys/vscode-solidity-metrics#12]
Can't create metrics for a specific file.
The message in the title is returned by VS Code.

**Comment by jacksanford1** [comment]
Having same problem. Have reloaded Solidity Metrics and restarted VS Code.

**Comment by hrishibhat** [comment]
> Having same problem. Right clicking on contracts folder. Have reloaded Solidity Metrics and restarted VS Code.

Were you able to sort this?

**Comment by dhruvtailor** [comment]
I am also facing the same issue has anyone found the solution?

**Comment by amateur-dev** [comment]
Same here

**Comment by tintinweb** [comment]
Hey, any chance you can help reproduce this issue? It does seem to work fine here on macos latest vscode.

thanks!

**Comment by zarifpour** [comment]
> Hey, any chance you can help reproduce this issue? It does seem to work fine here on macos latest vscode.
> 
> thanks!

It seems to be reproducible it in a workspace that contains _**multiple**_ projects/repositories. However, It **_does_** work as expected when I try it in a workspace that only contains a single project.

**Comment by jacksanford1** [comment]
Thanks for fixing @tintinweb 🙏

Has made my life a lot easier (and probably @hrishibhat's too).

**Comment by hrishibhat** [comment]
@jacksanford1 you bet! ;)

Thanks @tintinweb

**Comment by 0xSenzel** [comment]
Hi @tintinweb , I met with the same error when working on VS Code, hardhat JS template. I tried typescript template and empty folder none of them work.
Surprisingly I tried it on Gitpod and it had no issues even on empty folder. I'm not sure what is the exactly missing packages or settings that cause this error, do we need to install any prior packages for it to work?

**Comment by 0xSenzel** [comment]
Update:
Turns out right clicking the file / folder > Solidity: Metrics doesn't work but using `ctrl+shift+p` in vscode works just fine.
Problem only exists on my VSCode and Gitpod is fine, I created multiple different hardhat folders there is one seems to have the same issue inside Gitpod but I've compared the files they are practically the same.

Let me know if I should open an issue and what details I should give to help you figure out the problem.
